initrd/etc/luks-functions

```diff
@@ -351,17 +353,56 @@ luks_reencrypt() {
 			echo -e "\nEnter the current LUKS Disk Recovery Key passphrase:"
 			read -r luks_current_Disk_Recovery_Key_passphrase
 			echo -n "$luks_current_Disk_Recovery_Key_passphrase" >/tmp/luks_current_Disk_Recovery_Key_passphrase
-			warn "Reencrypting "$LUKS" LUKS encrypted drive content with a new LUKS Disk Recovery Key. Do NOT shut down or reboot!"
-			cryptsetup-reencrypt -B 64 --use-directio "$LUKS" --key-slot 0 --key-file /tmp/luks_current_Disk_Recovery_Key_passphrase
 		else
 			echo -n "$luks_current_Disk_Recovery_Key_passphrase" >/tmp/luks_current_Disk_Recovery_Key_passphrase
-			warn "Reencrypting "$LUKS" LUKS encrypted drive content with a new LUKS Disk Recovery Key. Do NOT shut down or reboot!"
-			cryptsetup-reencrypt -B 64 --use-directio "$LUKS" --key-slot 0 --key-file /tmp/luks_current_Disk_Recovery_Key_passphrase
 		fi
-		#Validate past cryptsetup-reencrypt attempts
-		if [ $(echo $?) -ne 0 ]; then
+
+		#Warn and launch actual reencryption
+		echo -e "\nReencrypting "$LUKS" LUKS encrypted drive content with current Recovery Disk Key passphrase..."
+		warn "DO NOT POWER DOWN MACHINE, UNPLUG AC OR REMOVE BATTERY DURING REENCRYPTION PROCESS"
+		# fist obtain which luks1/luks2 key-slot can be unlocked with the key-file
+		DRK_KEYSLOT=-1
+		for i in $(seq 0 7); do
```

Comment:
@UndeadDevel nice catch yes. This will also depend on checking luksDump "Version:" output and then be conditional. My intention is to use the last keyslot available per luks version. And as said in previous reply, wipe everything not being DRK and tested good for passphrase.

Thanks for your comments btw, always good to be reviewed and challenged. Props and thank you for your time!